b2sdk/file_version.py

```diff
@@ -101,6 +106,14 @@ def format_folder_ls_entry(cls, name):
         """ legacy method, to be removed in v2: formats a `ls` "folder" consistently with format_ls_entry() """
         return cls.LS_ENTRY_TEMPLATE % ('-', '-', '-', '-', 0, name)
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
```

Comment:
```suggestion
        if not isinstance(other, self.__class__):
            return False
```
I don't think we should check this. If it looks like a duck and quacks like a duck, lets treat it as a duck.